modules/account-lib/src/coin/sol/transaction.ts

```diff
@@ -266,16 +282,16 @@ export class Transaction extends BaseTransaction {
         'fee',
         'memo',
       ],
-      // TODO (STLX-8791): Need to figure out what id to use for unsigned transactions.
-      id: this.id || 'undefined',
+      id: this.id,
       type: TransactionType[this.type].toString(),
       changeOutputs: [],
       changeAmount: '0',
       outputAmount: outputAmount.toFixed(0),
       outputs: outputs,
       fee: {
-        // TODO(STLX-8791): need to figure out how to get fee information.
-        fee: '0',
+        fee: feeString,
+        feeRate: this.lamportsPerSignature,
+        type: 'perSignature',
```

Comment:
Where is this type field?

Reply:
it comes from the base fee: `baseCoin/iface.ts`:
```
export interface TransactionFee {
  fee: string;
  feeRate?: number;
  size?: number;
  type?: string;
}
```

Reply:
now that you pointed this out, type should probably be 'base'.
going to remove this for now since this is an optional field.